src/ui/public/state_management/state.js

```diff
@@ -205,7 +205,11 @@ export function StateProvider(Private, $rootScope, $location, config, kbnUrl) {
    */
   State.prototype._parseQueryParamValue = function (queryParam) {
     if (!isStateHash(queryParam)) {
```

Comment:
I think I agree, I suggest moving to that and we can take another look at it then